### PR TITLE
Add event before login

### PR DESCRIPTION
**Pull request by Jose96GIT**
I've added a call to apostrophe-login:before event as in the web login, so that when we add a behaviour in the event, it affects to both, web and api logins.

### CODE REVIEW

**Review by boutell — APPROVED** [review]
Good idea!

**Comment by boutell** [comment]
Hmm, actually I'm a bit concerned this could go wrong. Code written for the beforeLogin event would probably assume it's OK to manipulate req.session or even redirect the user, neither of which would go well in a headless context.

I think a separate event would be a wise choice for 2.x.

**Review by boutell — CHANGES_REQUESTED** [review]
Should be a separate event just for headless. Which means it can use `self.emit` rather than `callAllAndEmit`.

**Comment by Jose96GIT** [comment]
As I have changed the event to a separate one which exclusive for headless, I've also change the event name from `before` to `beforeLogin`. Do you agree with this name?

File: index.js [hunk 1]
```diff
@@ -60,6 +61,10 @@ module.exports = {
               return res.send({ bearer: bearer });
             }
           });
+          async function emitEventBeforeLogin(callback) {
+            await self.emit('beforeLogin', req);
```
Comment (boutell) [inline index.js:65]: This may throw an exception, in which case it must be caught and passed to the callback as the first argument.

**Review by boutell — CHANGES_REQUESTED** [review]

**Review by boutell — CHANGES_REQUESTED** [review]
Looks good — would you mind adding it to the README? Also please add a semicolon after `return callback(e)` although we could clean that up on our end if you prefer.

**Comment by Jose96GIT** [comment]
In addition to adding it to the README, I have also added it to the CHANGELOG.

**Review by boutell — APPROVED** [review]

**Comment by boutell** [comment]
Thanks!

**Comment by abea** [comment]
We're going to need to revert this. It is breaking failing several tests.

**Pull request merged**